sdk/lib/opentelemetry/sdk/resources/resource.rb

```diff
@@ -15,15 +15,16 @@ class << self
 
           # Returns a newly created {Resource} with the specified labels
           #
-          # @param [Hash<String, String>] labels Hash of key-value pairs to be used
+          # @param [Hash{String=>String, Numeric, Boolean>} labels Hash of key-value pairs to be used
```

Comment:
I think we have been taking the easy way out by using `Hash<String, Object>`, but this is more accurate and an established convention according to the [yardoc site](https://yardoc.org/types#see_results). @fbogsany, should we go forward with this and change our other docs accordingly. Follow up question, if so, should we do that as part of this PR? Followup to the followup, eventually we'll be supporting array value attributes, which will level up the documentation complications, should we worry about this, or am I getting too far ahead.

Reply:
If we want to update the other docs in the same way, I'm happy to do that next.

Reply:
That sounds good @fhwang. Do you want to do that as part of this PR or as a followup?

Also, taking a second look at this doc change, it does appear as though we have a lingering `>` and we should probably add spaces around the `=>`. Can we change this to `[Hash{String => String, Numeric, Boolean}]`?

Reply:
I suppose I'll do that work as a followup PR, in the interest of keeping PRs small when possible.

I've made the requested changes.